### ISSUE TITLE
[Bug][Firefox][macOS 13.3] Visiting any Google sites makes Dark Reader (Auto Mode) believe the whole system is in Light Mode after a few seconds

### ISSUE THREAD

**Issue by Write** [darkreader/darkreader#11247]
### Prerequisites

- [X] I [searched for any existing report](https://github.com/darkreader/darkreader/issues?q=is%3Aissue) about this bug to avoid opening a duplicate.
- [X] I can reproduce this bug in a new, unmodified web browser profile with Dark Reader installed as the only extension.
- [X] I understand I need to use the [Broken Website Report template](https://github.com/darkreader/darkreader/issues/new?assignees=&labels=Broken+Website&template=broken-website-report.yml&title=%5BBroken+Website%5D+Replace+with+title) if this bug I am reporting occurs on a single website.

### Bug Description

When visiting any Google websites, if the system is in dark mode and DarkReader is set to "Auto" ; after a few seconds on the website (for example, google search results), Dark Reader set itself to light mode, and says that the whole system is in Light Mode, when it's not. 

I couldn't find any other website having such behavior. 

It does seems that having Google Theme Parameter set on "Match color system" seems to almost fix the issue, unfortunately if you happen to visit another Google website (for example Google Photos) that doesn't respect this (Photos is always in light mode), the bug happens again. 

I tried on a new Firefox profile.

The behavior does not happen on Google Chrome. 

### Website Address

https://google.com

### Steps To Reproduce

1. Create a new Firefox profile in about:profiles for testing purposes
2. Download Dark Reader
3. Set to "Use System Color Scheme"
4. Have your system in dark color scheme
5. Go on Google or even Google Photos
6. Wait a few seconds
7. Web page turns in light mode (system is still in dark mode)
8. Open Dark Reader, and find that it says "Inactive, Auto (System is in light mode)"

It does seems that having Google Theme Parameter set on "Match color system" seems to almost fix the issue, unfortunately if you happen to visit another Google website (for example Google Photos) that doesn't respect this (Photos is always in light mode), the bug happens again. 

### Expected Behavior

Dark Reader should not believe the system color scheme is in light mode when Dark Reader is set in Auto Mode, and the system is in Dark Mode when visiting any Google website.

### Actual Behavior

Dark Reader in Auto Mode suddenly believe that the whole system's color scheme is in light mode when visting any Google website, after a few seconds, when the system is indeed still in dark mode.
I couldn't find any other website having such behavior.

### Operating System

macOS Ventura 13.3

### Web Browser name and version

Firefox 112.0.2

### Dark Reader version

4.9.63

**Comment by shibotto** [comment]
I get this also when hovering the app button (the 9 dots next to the login button). The breakage seems to occur when https://ogs.google.com gets loaded.

**Comment by Write** [comment]
> I get this also when hovering the app button (the 9 dots next to the login button). The breakage seems to occur when https://ogs.google.com gets loaded.

Tysm ! I blocked ogs.google.com at the DNS level it fixed my issue, it was driving me crazy since months. 
Ofc it'qs not a good idea if you love the app buttons, but it such a relief for me, pheww

**Comment by FranklinYu** [comment]
Thanks! Blocking OGS did the trick. I found that the request URL looks like

https://ogs.google.com/u/0/widget/app?origin=https%3A%2F%2Fwww.google.com&cn=app

It is loaded via `<iframe>`, but hidden until users click on the “Google Apps” button (also known as the “Google Waffle”).

**Comment by andreynering** [comment]
This happens to me as well.

**Comment by andreynering** [comment]
Looks like blocking that URL using uBlock Origin works as well.

Just a matter of adding this line:

```
ogs.google.com/u/0/widget/app
```

**Comment by Write** [comment]
> Looks like blocking that URL using uBlock Origin works as well.
> 
> Just a matter of adding this line:
> 
> ```
> ogs.google.com/u/0/widget/app
> ```
> 
> That specific widget stops working, of course. But I don't really use it.

Thanks, will unblock from my DNS for now. Better solution.

**Comment by shibotto** [comment]
I did a few testing and the problem starts with Firefox 106, while 105 and below seem OK. Dark Reader developers, you might want to start investigating in that direction.

Another website I find this issue in is https://docs.gaphor.org/en/latest/index.html, when the subdocument

```
https://utteranc.es/utterances.html?type=text%2Fjavascript&src=https%3A%2F%2Futteranc.es%2Fclient.js&async=&repo=gaphor%2Fgaphor-utterances&issue-term=pathname&theme=github-dark&crossorigin=anonymous&url=https%3A%2F%2Fdocs.gaphor.org%2Fen%2Flatest%2Findex.html&origin=https%3A%2F%2Fdocs.gaphor.org&pathname=en%2Flatest%2Findex&title=Gaphor+documentation&description=&og%3Atitle=&session=
```

gets loaded, similar to Google behavior. Don't be fooled by the default dark CSS of that website, look at Dark Reader icon or any other tab you have previously loaded.

**Comment by andreynering** [comment]
Thanks @Write.

Unfortunately I do use the account switcher because I use multiple accounts. Hopefully this bug will be fixed soon and we won't need the workaround anymore.

I've even duplicated the rule to include the other possible accounts:

```
||ogs.google.com/u/0/widget/app^
||ogs.google.com/u/1/widget/app^
||ogs.google.com/u/2/widget/app^
||ogs.google.com/u/3/widget/app^
||ogs.google.com/u/4/widget/app^
||ogs.google.com/u/5/widget/app^
```

**Comment by FranklinYu** [comment]
I can reproduce the issue on the Gaphor website. As @shibotto mentioned, the webpage looks dark, but the extension icon turned light.

**Comment by Write** [comment]
> For people only following this report:
> 
> More recent report that somehow superseded this: #11430 Fix in DarkReader 4.9.64 (#11254)
> 
> I confirm the fix, if I'm not the only one I guess this issue could be closed too.

Thanks a bunch, would've totally missed that. I removed the lines from uBlock Origin and can confirm it's fixed for me.
